Report: denser styling

paulirish opened the pull request:
This PR takes over from #2297 

Basic before and after: 
![image](https://user-images.githubusercontent.com/39191/28293695-51410d70-6b0a-11e7-8f4c-897ca0f3de38.png)

But let's do it with a few reports, because its all about the details:

### original:
http://dense-lh-reports.surge.sh/verge.orig.html
http://dense-lh-reports.surge.sh/omroep.orig.html
http://dense-lh-reports.surge.sh/paulirish.orig.html
http://dense-lh-reports.surge.sh/exampleperf.orig.html
http://dense-lh-reports.surge.sh/polymerperf.orig.html

### new look:
http://dense-lh-reports.surge.sh/verge.html
http://dense-lh-reports.surge.sh/omroep.html
http://dense-lh-reports.surge.sh/paulirish.html
http://dense-lh-reports.surge.sh/exampleperf.html
http://dense-lh-reports.surge.sh/polymerperf.html

### denser still, for devtools:
http://dense-lh-reports.surge.sh/verge.devtools.html
http://dense-lh-reports.surge.sh/omroep.devtools.html
http://dense-lh-reports.surge.sh/paulirish.devtools.html
http://dense-lh-reports.surge.sh/exampleperf.devtools.html
http://dense-lh-reports.surge.sh/polymerperf.devtools.html

patrickhulce commented on lighthouse-core/test/report/v2/renderer/category-renderer-test.js:
```diff
@@ -94,9 +94,7 @@ describe('CategoryRenderer', () => {
-    const audits = categoryDOM.querySelectorAll('.lh-category > .lh-audit, ' +
-        '.lh-category > .lh-passed-audits > .lh-audit, ' +
-        '.lh-audit-group--manual .lh-audit');
+    const audits = categoryDOM.querySelectorAll('.lh-audit');
```
❤️

patrickhulce commented on lighthouse-core/report/v2/templates.html:
```diff
@@ -265,14 +287,14 @@ <h1 class="leftnav__header__title">Lighthouse</h1>
       display: none;
     }
-    @media screen and (min-width: 1130px) {
+    @media screen and (min-width: 960px) {
```
this meant to be 964px?

paulirish replied:
yeah sorted these out. most are 964. there's a 965 in there too. 
thx!

paulirish commented:
> the debug string feels really close to the description even in non-devtools
sg. added 3px margin in all cases. feels good.

> should the description text be a bit smaller too?
changed it from 12px to 11px in devtools. (this was chris's original size, too)

>    disclosure triangles different colors
good call. sorted.

brendankenny commented on lighthouse-core/report/v2/renderer/category-renderer.js:
```diff
@@ -489,6 +490,11 @@ class CategoryRenderer {
     return element;
   }
+
+  _createPermalinkSpan(element, id) {
```
jsdocs for these to pass the closure check

brendankenny commented on lighthouse-core/report/v2/renderer/category-renderer.js:
```diff
@@ -210,28 +205,30 @@ class CategoryRenderer {
    * directly to the returned element.
    * @param {!ReportRenderer.GroupJSON} group
+   * @param {{expandable: boolean}} opts
    * @return {!HTMLDetailsElement}
```
should drop `HTMLDetailsElement` for `Element` (which breaks `el.open` but old-type-inference closure doesn't  catch that)

brendankenny commented on lighthouse-core/report/v2/renderer/category-renderer.js:
```diff
@@ -210,28 +205,30 @@ class CategoryRenderer {
+    const expandable = opts.expandable;
+    const auditGroupElem = /** @type {!HTMLDetailsElement} */ (
+          this._dom.createElement(expandable ? 'details' :'div',
           'lh-audit-group lh-expandable-details'));
```
Since this is shared by expandable and non-expandable, `lh-expandable-details` -> `lh-group` (or something better)? Is `lh-audit-group` only used for these?

paulirish replied:
sg. stripped expandable details from these.

brendankenny commented on lighthouse-core/report/v2/renderer/category-renderer.js:
```diff
@@ -210,28 +205,30 @@ class CategoryRenderer {
           'lh-audit-group lh-expandable-details'));
-    const auditGroupHeader = this._dom.createElement('div',
+    const auditGroupSummary = this._dom.createChildOf(auditGroupElem, 'summary',
+          'lh-audit-group__summary lh-expandable-details__summary');
```
`lh-expandable-details__summary`

brendankenny commented on lighthouse-core/report/v2/renderer/category-renderer.js:
```diff
@@ -210,28 +205,30 @@ class CategoryRenderer {
+    const auditGroupSummary = this._dom.createChildOf(auditGroupElem, 'summary',
+          'lh-audit-group__summary lh-expandable-details__summary');
+    const auditGroupHeader = this._dom.createChildOf(auditGroupSummary, 'div',
           'lh-audit-group__header lh-expandable-details__header');
```
lh-expandable-details__header

brendankenny commented on lighthouse-core/report/v2/renderer/category-renderer.js:
```diff
@@ -210,28 +205,30 @@ class CategoryRenderer {
-    });
-    auditGroupSummary.appendChild(auditGroupHeader);
-    auditGroupSummary.appendChild(auditGroupArrow);
+    if (group.description) {
```
another option is to put these as two templates, one expandable and one unexpandable. `_renderAuditGroup` could then just query the one it needs and then add a `description` as below if needed

brendankenny commented on lighthouse-core/report/v2/renderer/category-renderer.js:
```diff
@@ -167,6 +163,8 @@ class CategoryRenderer {
     const titleEl = this._dom.createChildOf(summary, 'div', 'lh-perf-hint__title');
     titleEl.textContent = audit.result.description;
 
+    this._dom.createChildOf(summary, 'div', 'lh-toggle-arrow', {title: 'See resources'});
```
can this be `lh-toggle-arrow-unexpandable`? Since it doesn't appear to expand?

paulirish replied:
This is the item level arrow in Opportunities, which is expandable:
![image](https://user-images.githubusercontent.com/39191/28389571-84a9687e-6c8b-11e7-8922-5ac4acd07121.png)

afaik, all non expandable arrows correctly have that class.

brendankenny commented:
lookin great

paulirish commented:
manually squashed and merged in 97c717030ef3194f0182a59d6a369d02e6d602b4
